pkg/controller/server/device/managed.go

```diff
@@ -167,7 +173,7 @@ func (e *external) Create(ctx context.Context, mg resource.Managed) (resource.Ex
 		return resource.ExternalCreation{}, errors.Wrap(err, errCreateDevice)
 	}
 
-	d.Status.ID = device.ID
+	d.Status.AtProvider.ID = device.ID
```

Comment:
It seems like we're recording this here because devices have a non-deterministic ID we need to use to refer to them. The new and improved pattern here is to:

1. Disable the default external naming behaviour by providing your own intializer chain per discussion in 
https://github.com/crossplaneio/crossplane-runtime/pull/45
1. Record this ID as the managed resource's external name annotation via `meta.SetExternalName()`
1. Use `meta.GetExternalName()` rather than `d.Status.AtProvider.ID` when making calls to `e.client`.

Reply:
Ah yes, thank you for pointing this out!